### PR TITLE
Update LocalProcessor to support CASE FeatHub expression

### PR DESCRIPTION
**Pull request by Sxnan**
## What is the purpose of the change

Update LocalProcessor to support CASE FeatHub expression. And Update LocalProcessor error handling.

## Brief change log

- Update LocalProcessor to support CASE FeatHub expression.
- Update LocalProcessor error handling.

## Does this pull request potentially affect one of the following parts:

  - Dependencies (does it add or upgrade a dependency): no
  - The public API: no

## Documentation

  - Does this pull request introduce a new feature? no
  - If yes, how is the feature documented? not applicable

### CODE REVIEW

**Comment by Sxnan** [comment]
@yunfengzhou-hub Could you review this PR?

File: python/feathub/processors/local/ast_evaluator/local_ast_evaluator.py [hunk 1]
```diff
@@ -152,10 +154,18 @@ def eval_group_node(self, ast: GroupNode, variables: Optional[Dict]) -> Any:
...
+        return left_value is None or np.isnan(left_value)
 
     def eval_null_node(self, ast: NullNode, variables: Optional[Dict]) -> Any:
-        raise RuntimeError("NULL operation is not supported.")
+        return None
 
     def eval_case_op(self, ast: CaseOp, variables: Optional[Dict]) -> Any:
-        raise RuntimeError("CASE operation is not supported.")
+        for condition_ast, result_ast in zip(ast.conditions, ast.results):
+            if self.eval(condition_ast, variables) is True:
```
Comment (yunfengzhou-hub) [inline python/feathub/processors/local/ast_evaluator/local_ast_evaluator.py:165]: Let's handle cases in which `self.eval(condition_ast, variables)` returns a value other than `True` and `False`. I found that some sql engines would throw an exception as follows.

> Case error: An expression of non-boolean type specified in a context where a condition is expected

**Review by yunfengzhou-hub — COMMENTED** [review]
Thanks for the PR. Left some comments as below.

**Comment by yunfengzhou-hub** [comment]
Thanks for the update. LGTM. @lindong28 could you please take a look at this PR?

**Comment by lindong28** [comment]
Thanks for the PR. LGTM.

**Pull request merged**